ibazel/main.go

```diff
@@ -94,27 +101,54 @@ func main() {
 		return
 	}
 
-	// Listen to the events and trigger action based on the response code.
-	go func() {
-		for {
+	state := QUERY
+	for {
+		fmt.Printf("State: %s\n", state)
+		switch state {
+		case WAIT:
+			select {
+			case <-sourceFileWatcher.Events:
+				fmt.Printf("Detected source change. Rebuilding...\n")
+				state = DEBOUNCE_RUN
+			case <-buildFileWatcher.Events:
+				fmt.Printf("Detected build graph change. Requerying...\n")
+				state = DEBOUNCE_QUERY
+			}
+		case DEBOUNCE_QUERY:
+			state = QUERY
+			select {
```

Comment:
I am going to make a wrapper around the watchers that denounces for me in the next CL and remove the debounce states.